app/models/course/lesson_plan/item.rb

```diff
@@ -152,25 +152,43 @@ def find_or_create_personal_time_for(course_user)
     personal_time
   end
 
-  # Finds the lesson plan items which are starting within the next day for a given course.
+  def self.upcoming_for_exists?(course)
+    opening_items = course.lesson_plan_items.published.
+                    eager_load(:personal_times, :reference_times).
+                    preload(:actable)
+    opening_items.select { |item| item.actable.include_in_consolidated_email?(:opening) }.each do |item|
+      course.course_users.each do |course_user|
+        return true if item.time_for(course_user).start_at.in?((Time.zone.now)..(1.day.from_now))
+      end
+    end
+    false
+  end
+
+  # Finds the lesson plan items which are starting within the next day for a given course user.
   # Rearrange the items into a hash keyed by the actable type as a string.
   # For example:
   # {
   #   ActableType_1_as_String => [ActableItems...],
   #   ActableType_2_as_String => [ActableItems...]
   # }
   #
-  # @param course [Course] The course to check for published items starting within the next day.
+  # @param course_user [CourseUser] The course user to check for published items starting within the next day.
   # @return [Hash]
-  def self.upcoming_items_from_course_by_type(course)
-    opening_items = course.lesson_plan_items.published.opening_within_next_day
```

Comment:
I think `opening_within_next_day` scope is not used by anything else, can remove

Reply:
Ok done